test/onnx/test_pytorch_onnx_no_runtime.py

```diff
@@ -2,19 +2,70 @@
 
 """Tests for onnx export that don't run the exported model."""
 
+import contextlib
 import io
+import itertools
 import unittest
-from typing import Dict, Optional, Type
+from typing import Dict, Optional, Type, Callable, Iterable, Tuple, Union
 
 import onnx
 
 import torch
 from torch import Tensor
-from torch.onnx import symbolic_helper, utils
+from torch.onnx import symbolic_helper, utils, symbolic_registry
+from torch.onnx._globals import GLOBALS
 from torch.testing._internal import common_utils
 
 
-class TestOptionalOutput(unittest.TestCase):
+def export_to_onnx(
+    model: Union[torch.nn.Module, torch.jit.ScriptFunction],
+    input: Tuple[torch.Tensor],
+    custom_ops: Optional[
+        Iterable[
+            Union[contextlib.AbstractContextManager, contextlib.ContextDecorator],
+        ]
+    ] = None,
+    mocks: Optional[Iterable] = None,
+    operator_export_type: torch.onnx.OperatorExportTypes = torch.onnx.OperatorExportTypes.ONNX,
+    opset_version: int = GLOBALS.export_onnx_opset_version,
+) -> onnx.ModelProto:
+    """Exports `model(input)` to ONNX and returns it.
+
+    Custom operators and/or unittest patches can be used help reproducing specific behaviors.
+
+    Args:
+        model: model to export
+        input: model input with same format as `torch.onnx.export(..,args,...)`
+        custom_ops: list of custom operators to use during export
+        mocks: list of mocks to use during export
+        operator_export_type: export type as described by `torch.onnx.export(...operator_export_type,...)`
+        opset_version: ONNX opset version as described by `torch.onnx.export(...opset_version,...)`
+    Returns:
+        A valid ONNX model (`onnx.ModelProto`)
+    """
+    custom_ops = custom_ops or []
+    mocks = mocks or []
+    with contextlib.ExitStack() as stack:
+        for ctx in itertools.chain(custom_ops, mocks):
+            stack.enter_context(ctx)
+
+        f = io.BytesIO()
+        torch.onnx.export(
+            model,
+            input,
+            f,
+            operator_export_type=operator_export_type,
+            opset_version=opset_version,
+        )
+
+    # Validate ONNX graph before returning it
+    onnx_model = onnx.load_from_string(f.getvalue())
+    onnx.checker.check_model(onnx_model)
+    return onnx_model
+
+
+@common_utils.instantiate_parametrized_tests
+class TestOptionalOutput(common_utils.TestCase):
     # TODO: Move these tests to test_pytorch_onnx_onnxruntime once
```

Comment:
can you please update this to reference a GitHub issue that tracks this? Can be in ORT or PyTorch

Reply:
https://github.com/microsoft/onnxruntime/issues/11717